tap_dashboard_reports/streams.py

```diff
@@ -49,23 +49,24 @@ def schema(self) -> dict:
         """Dynamically detect the json schema for the stream.
         This is evaluated prior to any records being retrieved.
         """
-
         properties = [
             th.Property(field, th.StringType)
             for field in (self._dimensions + self._measures)
         ]
-
+        if len(self.primary_keys) != 0:
```

Comment:
I believe this logic should in primary_keys method 😄. 
And now you do not use your param from yml file

Reply:
`primary_keys` will be always > 0. You should your param from yml file

Reply:
- already takes from yml file
- creating custom_key flag